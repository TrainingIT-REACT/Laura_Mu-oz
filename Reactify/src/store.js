import { createStore, combineReducers } from 'redux';

import myList from './reducers/myList';
import user from './reducers/user';

const store = createStore(
    combineReducers({
        myList,
        user
    })
)

export default store;