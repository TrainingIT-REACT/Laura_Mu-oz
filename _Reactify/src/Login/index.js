import React, { createRef } from 'react';
import { connect } from 'react-redux';
import {updateUsername} from '../actions/user';
import {clearList} from '../actions/myList';
import store from '../store';
import "@babel/polyfill";


const Login = ({updateUsername, clearList}) =>{
    const input = createRef();
    const log = createRef();
    console.log(store.getState());
    const onSubmit = (e) => {
        e.preventDefault();
        updateUsername(input.current.value);
        log.current.style.display = "none";
    }

        if(store.getState().user.name === ""){
            if(window.location.pathname !== '/recently'){
                return(
                    <div>
                        <h1>Login page </h1>
                        <form ref={log} onSubmit={onSubmit}>
                            <div class="form-group">
                                <label for="exampleInputEmail1">Username</label>
                                <input type="text" ref={input} class="form-control" id="exampleInputEmail1" aria-describedby="emailHelp" placeholder="Enter email" />
                            </div>
                            <div class="form-group">
                                <label for="exampleInputPassword1">Password</label>
                                <input type="password" class="form-control" id="exampleInputPassword1" placeholder="Password" />
                            </div>
                            <button>Login</button>
                        </form>
                    </div>
                );
            }else{
                return(
                    <div>
                        <h1>Login page </h1>
                        <form ref={log} onSubmit={onSubmit}>
                            <div class="form-group">
                                <label for="exampleInputEmail1">Username</label>
                                <input type="text" ref={input} class="form-control" id="exampleInputEmail1" aria-describedby="emailHelp" placeholder="Enter email" />
                            </div>
                            <div class="form-group">
                                <label for="exampleInputPassword1">Password</label>
                                <input type="password" class="form-control" id="exampleInputPassword1" placeholder="Password" />
                            </div>
                            <button>Login</button>
                        </form>
                        <div style={{marginTop: "8px"}} className="alert alert-danger">You should be loged to watch out your recents songs</div>
                    </div>
                );
            }
        }else{
            return(
                <div>
                    <h3 ref={input}>Â¡Bienvenido {store.getState().user.name}!</h3>
                        <button onClick={() => {updateUsername(""); clearList()}}>Logout</button>
                </div>
            );
        }
}

const mapDispatchToProps = (dispatch) => ({
    updateUsername: (name) => dispatch(updateUsername(name)),
    clearList: () => dispatch(clearList())
  });

export default connect(
    () => ({}),
    mapDispatchToProps,
  )(Login);