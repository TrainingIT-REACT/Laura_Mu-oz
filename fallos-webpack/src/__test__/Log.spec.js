import React from 'react';
import { Provider } from "react-redux";
import { render } from 'enzyme';
import Login from '../Login';
import store from '../store';

describe("Login", () => {
  describe( "render", () => {
    it("should render all HTML elements", () => {
      const wrapper = render(<Provider store={store}><Login /></Provider> );
      expect(wrapper.find('label').length).toBe(2);
      expect(wrapper.find('button').length).toBe(1);
      expect(wrapper.find('form')).toBeTruthy();
      expect(wrapper).toMatchSnapshot();
    })
  })
})