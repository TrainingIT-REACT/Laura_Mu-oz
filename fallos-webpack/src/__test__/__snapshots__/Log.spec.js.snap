// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login render should render all HTML elements 1`] = `
<div>
  <h1>
    Login page 
  </h1>
  <form>
    <div
      class="form-group"
    >
      <label
        for="exampleInputEmail1"
      >
        Username
      </label>
      <input
        aria-describedby="emailHelp"
        class="form-control"
        id="exampleInputEmail1"
        placeholder="Enter email"
        type="text"
      />
    </div>
    <div
      class="form-group"
    >
      <label
        for="exampleInputPassword1"
      >
        Password
      </label>
      <input
        class="form-control"
        id="exampleInputPassword1"
        placeholder="Password"
        type="password"
      />
    </div>
    <button>
      Login
    </button>
  </form>
</div>
`;
